File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/UserModel.java
package com.example.simplehabitassignment.data.vos.models;


import com.example.simplehabitassignment.data.vos.LoginUserVO;
import com.example.simplehabitassignment.delegates.LoginDelegate;

public interface UserModel {

    void login(String emailOrPassword, String password, LoginDelegate loginDelegate);


    void onUserLogout();

    LoginUserVO getLoginUser();



    boolean isUserdLogin();
}

File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/UserModel.kt
package com.example.simplehabitassignment.data.vos.models


import com.example.simplehabitassignment.data.vos.LoginUserVO
import com.example.simplehabitassignment.delegates.LoginDelegate

interface UserModel {

    val loginUser: LoginUserVO


    val isUserdLogin: Boolean

    fun login(emailOrPassword: String, password: String, loginDelegate: LoginDelegate)


    fun onUserLogout()
}
